### PR TITLE
Multiple Kanban fixes

### PR DESCRIPTION
**Pull request by mdroidian**
https://www.loom.com/share/cc48bec4e512487ea040186ef99cbc62

### Issues Fixed

- Initial Column Settings
- Display tasks if "No {columnKey}"

image

- Being able to select "text" (first column) as Group By or Display

image

### CODE REVIEW

File: src/components/Kanban.tsx [hunk 1]
```diff
@@ -287,105 +322,128 @@ const Kanban = ({
...
+          <div className="inline-block mr-4">
+            <span className="font-bold">Group By:</span>
+            <span> {columnKey}</span>
+          </div>
+          <div className="inline-block">
+            <span className="font-bold">Display:</span>
+            <span> {displayKey}</span>
+          </div>
+        </div>
+      )}
```
Comment (mdroidian) [inline src/components/Kanban.tsx:347]: this is the only actual change in the `return`, the rest is from the `<></>`

File: src/components/Kanban.tsx [hunk 2]
```diff
@@ -156,7 +159,15 @@ const Kanban = ({
       prev[key] += 1;
       return prev;
     }, {} as Record<string, number>);
-    return Object.entries(valueCounts)
+    const cleanedValueCounts = Object.entries(valueCounts).reduce(
```
Comment (dvargas92495) [inline src/components/Kanban.tsx:162]: try to use `map` over `reduce` where possible

File: src/components/Kanban.tsx [hunk 3]
```diff
@@ -191,11 +202,12 @@ const Kanban = ({
...
-        value: d[columnKey],
-        defaultValue: `No ${columnKey}`,
-        uid: d[`${columnKey}-uid`]?.toString(),
-      });
+      const column =
+        toCellValue({
+          value: d[columnKey],
+          defaultValue: `No ${columnKey}`,
+          uid: d[`${columnKey}-uid`]?.toString(),
+        }) || `No ${columnKey}`;
```
Comment (dvargas92495) [inline src/components/Kanban.tsx:210]: I'd define a `DEFAULT_FORMAT` then interpolate the `columnKey` in there. Someone will eventually ask to configure it

File: src/components/Kanban.tsx [hunk 4]
```diff
@@ -274,11 +287,33 @@ const Kanban = ({
...
+      if (isRemoveValue && !previousValue) return;
       if (typeof columnUid !== "string") return;
+      if (!columnUid) {
+        createBlock({
+          parentUid: result.uid,
+          node: {
+            text: `${columnKeySelection}:: ${column}`,
+          },
+        }).then(onQuery);
+        return;
```
Comment (dvargas92495) [inline src/components/Kanban.tsx:304]: move this into `predefinedSelection` with a flag that says `create: true` or something like that

File: src/components/Kanban.tsx [hunk 5]
```diff
@@ -274,11 +287,33 @@ const Kanban = ({
...
+      }
+      if (!previousValue) {
+        updateBlock({
+          uid: columnUid,
+          text: `${columnKeySelection}:: ${column}`,
+        }).then(onQuery);
+        return;
+      }
```
Comment (dvargas92495) [inline src/components/Kanban.tsx:312]: send `previousValue` as a prop too

**Review by dvargas92495 — COMMENTED** [review]
Moving into `update` is the only blocking comment

**Pull request merged**